python/ray/tests/accelerators/test_hpu.py

```diff
@@ -0,0 +1,118 @@
+    return_value=4,
+)
+def test_auto_detected_more_than_visible(
+    mock_get_num_accelerators, monkeypatch, shutdown_only
+):
+    # Test more hpus are detected than visible.
+    monkeypatch.setenv("HABANA_VISIBLE_MODULES", "0,1,2")
+    ray.init()
+    mock_get_num_accelerators.called
+    assert ray.available_resources()["HPU"] == 3
+
+
+@patch(
+    "ray._private.accelerators.HPUAcceleratorManager.get_current_node_num_accelerators",  # noqa: E501
+    return_value=2,
+)
+def test_auto_detect_resources(mock_get_num_accelerators, shutdown_only):
+    # Test that ray node resources are filled with auto detected count.
+    ray.init()
+    mock_get_num_accelerators.called
+    assert ray.available_resources()["HPU"] == 2
+
+
+def test_get_current_process_visible_accelerator_ids():
+    os.environ[hpu.HABANA_VISIBLE_DEVICES_ENV_VAR] = "0,1,2"
+    assert HPUAcceleratorManager.get_current_process_visible_accelerator_ids() == [
+        "0",
+        "1",
+        "2",
+    ]  # noqa: E501
+
+    del os.environ[hpu.HABANA_VISIBLE_DEVICES_ENV_VAR]
+    assert HPUAcceleratorManager.get_current_process_visible_accelerator_ids() is None
+
+    os.environ[hpu.HABANA_VISIBLE_DEVICES_ENV_VAR] = ""
+    assert HPUAcceleratorManager.get_current_process_visible_accelerator_ids() == []
+
+    del os.environ[hpu.HABANA_VISIBLE_DEVICES_ENV_VAR]
+
+
+def test_set_current_process_visible_accelerator_ids():
+    HPUAcceleratorManager.set_current_process_visible_accelerator_ids(["0"])
+    assert os.environ[hpu.HABANA_VISIBLE_DEVICES_ENV_VAR] == "0"
+
+    HPUAcceleratorManager.set_current_process_visible_accelerator_ids(["0", "1"])
+    assert os.environ[hpu.HABANA_VISIBLE_DEVICES_ENV_VAR] == "0,1"
+
+    HPUAcceleratorManager.set_current_process_visible_accelerator_ids(["0", "1", "2"])
+    assert os.environ[hpu.HABANA_VISIBLE_DEVICES_ENV_VAR] == "0,1,2"
+
+    del os.environ[hpu.HABANA_VISIBLE_DEVICES_ENV_VAR]
+
+
+@pytest.mark.parametrize(
+    "test_config",
+    [
+        (1, False),
+        (0.5, True),
+        (3, False),
+    ],
+)
+def test_validate_resource_request_quantity(test_config):
+    num_hpus, expect_error = test_config
+
+    if expect_error:
+        assert (
+            HPUAcceleratorManager.validate_resource_request_quantity(num_hpus)[0]
+            is False
+        )
+        assert (
+            HPUAcceleratorManager.validate_resource_request_quantity(num_hpus)[1]
+            is not None
+        )
+    else:
+        assert (
+            HPUAcceleratorManager.validate_resource_request_quantity(num_hpus)[0]
+            is True
+        )
+        assert (
+            HPUAcceleratorManager.validate_resource_request_quantity(num_hpus)[1]
+            is None
+        )
+
+
+def test_check_accelerator_info():
+
+    if hpu.HPU_PACKAGE_AVAILABLE:
+        assert "GAUDI" in HPUAcceleratorManager.get_current_node_accelerator_type()
```

Comment:
This is wrong?

Reply:
This is not wrong but will be fruitful to add an initialization check here.